components/Sidebar/panels/Friends.tsx

```diff
@@ -1,82 +1,88 @@
+import { doc, getDoc } from "firebase/firestore";
+import Link from "next/link";
+import { notFound } from "next/navigation";
 import "server-only";
 
 import UsernameLink from "@/components/ui/UsernameLink";
+import { profilesCol } from "@/lib/firestore";
+import getUsersFollowing from "@/lib/followers/getUsersFollowing";
+import isUserFollowing from "@/lib/followers/isUserFollowing";
 
 import BasePanel from "./BasePanel";
 
-const MOCK_DATA = [
-  {
-    id: "1",
-    title: "Kevin",
-    username: "kevinnguyen",
-    timestamp: "1 hr",
-    body: "Colors - Thunder Jackson",
-  },
-  {
-    id: "2",
-    title: "Sophie",
-    username: "sophie-saffron",
-    timestamp: "10 hr",
-    body: "Shotgun - George Ezra",
-  },
-  {
-    id: "3",
-    title: "Dalton",
-    username: "cravend123",
-    timestamp: "12 hr",
-    body: "Les autres on verra - Madam Monsieur",
-  },
-  {
-    id: "4",
-    title: "Mason",
-    username: "dkms",
-    timestamp: "21 hr",
-    body: "Tyler Herro - Jack Harlow",
-  },
-];
+const getTopItems = async (id: string) => {
+  const profileDoc = await getDoc(doc(profilesCol, id));
+  if (!profileDoc.exists()) {
+    notFound();
+  }
 
-const Friend = ({
-  title,
-  timestamp,
+  return profileDoc.data();
+};
+
+const Friend = async ({
   username,
-  body,
+  userId,
 }: {
-  title: string;
-  timestamp: string;
-  username: string | undefined;
-  body: string;
-}) => (
-  <li>
-    <div className="h-fit">
-      <div className="flex flex-row justify-between items-center">
-        <h4 className="normal-case font-bold">
-          {username ? (
-            <UsernameLink username={username}>{title}</UsernameLink>
-          ) : (
-            title
-          )}
-        </h4>
-        <h5 className="normal-case font-bold">{timestamp}</h5>
+  username: string;
+  userId: string;
+}) => {
+  const data = await getTopItems(userId);
+  const tracks = data.topTracks.slice(0, 2);
+  return (
+    <div>
+      <div className="h-fit">
+        <div className="flex flex-row justify-between items-center">
+          <h4 className="normal-case font-bold">
+            {username ? (
+              <UsernameLink username={username}>{username}</UsernameLink>
+            ) : (
+              username
+            )}{" "}
+            is listening to
+          </h4>
+        </div>
+        <div>
```

Comment:
Can you add some styles to this to make it look better? So the words aren't all spaced out and stuff